lang/py3/avro/tests/gen_interop_data.py

```diff
@@ -46,11 +47,18 @@
 }
 
 
-if __name__ == "__main__":
-  interop_schema = schema.Parse(open(sys.argv[1], 'r').read())
-  writer = open(sys.argv[2], 'wb')
+def generate(schema_file, output_path):
+  interop_schema = schema.Parse(open(schema_file, 'r').read())
   datum_writer = io.DatumWriter()
-  # NB: not using compression
-  dfw = datafile.DataFileWriter(writer, datum_writer, interop_schema)
-  dfw.append(DATUM)
-  dfw.close()
+  for codec in datafile.VALID_CODECS:
+    filename = 'py3'
+    if codec != 'null':
+      filename += '_' + codec
+    with Path(output_path, filename).with_suffix('.avro').open('wb') as writer:
+      dfw = datafile.DataFileWriter(writer, datum_writer, interop_schema, codec)
```

Comment:
A DataFileWriter is a context manager too.

Reply:
Oh I missed that. Just fixed it, thanks @kojiromike!